docs/spark/README.md

```diff
@@ -0,0 +1,122 @@
+
+## Table of Contents
+
+* [Spark Components](#spark-components)
+  * [Master](#master)
+  * [Worker](#workers-and-executors)
+  * [Driver](#driver)
+* [Development Environment](#development-environment)
+  * [Building a Job JAR](#building-a-job-jar)
+  * [Local Spark Standalone Cluster](#local-spark-standalone-cluster)
+
+## Spark Components
+
+Often times Spark is deployed on top of a cluster resource manager, such as Apache Mesos or Apache Hadoop YARN (Yet Another Resource Negotiator). In addition to those options, Spark comes bundled with **Spark Standalone**, which is a built-in way to run Spark in a clustered environment.
+
+A Spark Standalone cluster has the following components:
+
+### Driver
+
+A Spark Standalone driver prepares the Spark context and submits an execution graph to the **master**. As progress is made against the jobs, the driver also helps coordinate across job stages.
+
+### Master
+
+The Spark Standalone master provides an RPC endpoint for **drivers** and **workers** to communicate with. It also creates tasks out of a job's execution graph and submits them to **workers**.
+
+There is generally only one master in a Spark Standalone cluster, but when multiple master are up at the same time one must be elected the leader.
+
+### Workers and Executors
+
+Workers within a Spark Standalone cluster interact with the **master** to determine what jobs they can do. Within a **worker** exist **executors**, and executors work on tasks within the jobs assigned to a worker. They also communicate task status back to the **driver**.
+
+A Spark Standalone cluster can have one or more workers within a cluster, and each worker can have one or more executors (generally limited to the number of CPUs available on the worker).
+
+## Development Environment
+
+The local development environment for Raster Foundry contains a Docker Compose configuration for running a Spark Standalone cluster. All services within the Docker Compose configuration share a common base image (only the **master** builds on that slightly to prepare the cluster state directory for recovery).
+
+### Building a Job JAR
+
+Jobs are submitted to a Spark Standalone cluster as a JAR file. In order to exercise the cluster, this repository contains a `SparkPi` job that can be compiled into a JAR with:
+
+```bash
+$ docker-compose \
+    -f docker-compose.spark.yml run --rm --no-deps --user root --entrypoint ./sbt \
+    spark-driver package
+```
+
+That process will produce `rf-worker_2.11-0.1.0.jar` inside of the `target/scala-2.11` directory.
+
+### Local Spark Standalone Cluster
+
+To get the Spark Standalone environment up-and-running, build the **master** container image:
+
+```bash
+$ docker-compose -f docker-compose.spark.yml build
+```
+
+Next, bring up the Spark master:
+
+```bash
+$ docker-compose -f docker-compose.spark.yml up spark-master
```

Comment:
I think a note may need to be added to signal that these steps are intended to be run from multiple terminal windows.

The main reason I don't want them to go into the background is that I want people to be able to interactively inspect the output when different types of failure scenarios are triggered.

Addressed in https://github.com/azavea/raster-foundry/pull/424/commits/56674c43f1229023d43922cb36e29c3439115349.